### PR TITLE
[query] pca simplification/optimization

### PR DESCRIPTION
**Pull request by patrick-schultz**
This flips the pca algorithm, so that we're locally collecting a Krylov space basis on the column side (presumably the smaller side). This leads to a few simplifications, and allows for an optimization in the `compute_loadings=False` case. Once we have a complete TSQR implementation, we can optimize the `compute_loadings=True` case as well. With that, there should be no memory limitation to the number of rows (no local value is O(rows)).

### CODE REVIEW

**Comment by patrick-schultz** [comment]
This PR broke `test_blanczos_against_hail`, but the difference between the two was only slightly above the threshold. This is because, for a fixed value of `q`, the new version effectively does half an iteration less than before. For example, for `q=2`, the old version computed a basis for `[AG, AA'AG, (AA')^2AG]`, while the new version computes a basis for `[G, A'AG, (A'A)^2G]`. Incrementing `q` by one in the test fixed it.

File: hail/python/hail/methods/pca.py [hunk 1]
```diff
@@ -315,64 +315,59 @@ def _blanczos_pca(entry_expr, k=10, compute_loadings=False, q_iterations=2, over
...
-            localized_H_i = hl.nd.vstack(result.Hi_chunks)
-            h_list.append(localized_H_i)
-            G_i = hl.nd.qr(result.G_i)[0]
-
-        info(f"blanczos_pca: Beginning iteration {q+ 1}/{q+1}")
-        temp = A.annotate(H_i=A.ndarray @ G_i)
-        result = temp.aggregate(hl.agg.collect(temp.H_i), _localize=False)._persist()
+            info(f"blanczos_pca: Beginning iteration {j+1}/{q}")
+            G_i = A.aggregate(hl.agg.ndarray_sum(A.ndarray.T @ (A.ndarray @ G_i)), _localize=False)
+            G_i = hl.nd.qr(G_i)[0]._persist()
```
Comment (johnc1231) [inline hail/python/hail/methods/pca.py:326]: Why persist individual `G_i`s? Seems like you use them each exactly once.

**Review by johnc1231 — CHANGES_REQUESTED** [review]

File: hail/python/hail/methods/pca.py (hunk 1 above)
Comment (patrick-schultz) [reply hail/python/hail/methods/pca.py:326]: We save them in `g_list` and use them in the next iteration.

Comment (johnc1231) [reply hail/python/hail/methods/pca.py:326]: Duh, you're right. Approved.

**Review by johnc1231 — APPROVED** [review]

**Pull request merged**